apps/api/src/app/workflows-v2/util/template-parser/liquid-parser.ts

```diff
@@ -213,3 +331,27 @@ function extractProps(template: any): { valid: boolean; props: string[]; error?:
 
   return { valid: true, props: validProps };
 }
+
+function extractFilters(template: Template): Filter[] {
+  if (template instanceof Output) {
+    return template.value.filters;
+  }
+
+  return [];
+}
+
+function validateFilters(filters: Filter[], isDigestEventsVariable: boolean): LiquidFilterIssue[] {
+  return filters.reduce((acc, filter) => {
+    const validator = FILTER_VALIDATORS[filter.name];
+    if (!validator) return acc;
+
+    let args: unknown[] = [...filter.args];
+    if (filter.name === 'toSentence') {
+      args = [{ requireKeyPath: isDigestEventsVariable }, ...filter.args];
+    }
```

Comment:
if the `toSentence` filter is used for the `steps.x.events` variable, then we want to validate that the `key path` is provided

Reply:
Don't we want to always require the key path?